modules/affile/affile-dest.c

```diff
@@ -769,14 +769,13 @@ affile_dd_new_instance(gchar *filename, GlobalConfig *cfg)
   self->super.super.super.queue = affile_dd_queue;
   self->super.super.super.free_fn = affile_dd_free;
   self->super.super.super.generate_persist_name = affile_dd_format_persist_name;
-  self->filename_template = log_template_new(cfg, NULL);
-  log_template_compile(self->filename_template, filename, NULL);
+  self->filename_template = filename_template;
   log_writer_options_defaults(&self->writer_options);
   self->writer_options.mark_mode = MM_NONE;
   self->writer_options.stats_level = STATS_LEVEL1;
   self->writer_flags = LW_FORMAT_FILE;
 
-  if (strchr(filename, '$') != NULL)
+  if (strchr(filename_template->template, '$') != NULL)
```

Comment:
This looks pretty bad (@Kokan noticed that too). After my change, the layering violation became clear.

I was planning to add a `log_template_is_literal_string()`/`log_template_get_literal()` pair to the template API for #3621.

I'll push the commit containing those methods into this branch instead, so we can replace the `strchr()` call with `log_template_is_literal_string()`, which makes sure the template is really a literal string in the appropriate layer (without using `strchr()`).